app/main.py

```diff
@@ -1,2 +1,101 @@
+from typing import Union
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.hash_table: list = [
+            [None, None],
+            [None, None],
+            [None, None],
+            [None, None],
+            [None, None],
+            [None, None],
+            [None, None],
+            [None, None]
+        ]
+
+    def __setitem__(self, key: Union, value: Union) -> None:
+        index_hash = self.hashed(key)
+        if self.hash_table[index_hash][0] is key:
+            self.hash_table[index_hash][1] = value
+            return
+        if (
+            self.hash_table.count([None, None])
+            == int(len(self.hash_table) / 3 + 1)
+        ):
+            copy_old_table = self.hash_table.copy()
+            self.hash_table = [[None, None]] * len(self.hash_table) * 2
+            for cell in range(len(copy_old_table)):
+                if copy_old_table[cell] != [None, None]:
+                    new_cell = self.hashed(self.hash_table[cell][0])
+                    self.adding_new_value(
+                        new_cell,
+                        copy_old_table[cell][0],
+                        copy_old_table[cell][1]
+                    )
+        if self.hash_table[index_hash][0] is None:
+            self.hash_table[index_hash][0] = key
+            self.hash_table[index_hash][1] = value
+            return
+        self.adding_new_value(10, key, value)
+
+    def adding_new_value(self, index_hash_table, key, value) -> None:
+        for _ in range(len(self.hash_table)):
```

Comment:
It must works with while loop:
while (here is condition with check that cell is available to fill, if not available continue loop)
inside loop just do:
index_hash_table = (index_hash_table + 1) % len(self.has_table)